core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -86,16 +82,13 @@ protected CachingCatalog(Catalog catalog, boolean caseSensitive, long expiration
   }
 
   /**
-   * CacheWriter class for removing metadata tables when their associated data table is expired
+   * RemovalListener class for removing metadata tables when their associated data table is expired
    * via cache expiration.
    */
-  class MetadataTableInvalidatingCacheWriter implements CacheWriter<TableIdentifier, Table> {
-    @Override
-    public void write(TableIdentifier tableIdentifier, Table table) {
-    }
-
+  class MetadataTableInvalidatingRemovalListener implements RemovalListener<TableIdentifier, Table> {
     @Override
-    public void delete(TableIdentifier tableIdentifier, Table table, RemovalCause cause) {
+    public void onRemoval(TableIdentifier tableIdentifier, Table table, RemovalCause cause) {
+      LOG.debug("Evicted {} from the table cache ({})", tableIdentifier, cause);
```

Comment:
Given that this inner class is the only thing that uses the Logger, should we make the call to LoggerFactory.getLogger here instead?

Reply:
static declarations are not supported in inner classes :(